### ISSUE TITLE
Different behaviour for boost_from_average

### ISSUE THREAD

**Issue by lunz-s** [microsoft/LightGBM#2513]
I have been provided with a LightGBM model from a third party that they have trained in R (presumably using LightGBM 2.1.12) alongside their training data set. I have been trying to reproduce their model, reading out their hyperparameters from the save and training my own LightGBM model in python (using version 2.3.0) on their training data, but failed to get the same leaf values.

In particular, both models are using boost_from_average=1, but this seems to induce different behaviour for me to what happened in their model. In my case, the shrinkage rate of the 0th tree is set to 1 and the offset coming from boost_from_average seems to be added to the leaf values. Their model does not seem to exhibit this type of behaviour, with the 0th shrinkage rate being the usual learning rate and the leaf values not including the class average offset. 

Does anyone have any idea what could be inducing this difference in behaviour? Is it a version conflict (https://github.com/microsoft/LightGBM/issues/1514)? Or might there be a hyperparameter that I am missing that changes the behaviour under boost_from_average? Thanks a lot for any help!

**Comment by guolinke** [comment]
2.1.12? or 2.1.2?

you can try to re-run it with that version, to check the behavior.

**Comment by StrikerRUS** [comment]
ping @lunz-s

**Comment by StrikerRUS** [comment]
I don't think that we had `2.1.12` version. Moreover, `2.2.0` was released *Sep 18 2018*. A lot of time has passed and a lot of changes have been made since that time. Unfortunately, we started to provide changelogs only from `2.3.0` version. So, the only way to figure it out is to dig deep into GitHub history, but I guess it will not bring any success due to huge gap between comparing version numbers.

https://github.com/microsoft/LightGBM/releases

Closing due to inactivity. Feel free to reopen with new info.
